src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -272,10 +273,27 @@ void ImageReplayer<I>::add_peer(const Peer<I>& peer) {
   dout(10) << "peer=" << peer << dendl;
 
   std::lock_guard locker{m_lock};
-  auto it = m_peers.find(peer);
-  if (it == m_peers.end()) {
-    m_peers.insert(peer);
+  auto result = m_peers.insert(peer).second;
+  ceph_assert(result);
```

Comment:
Shouldn't be `add_peer` updated to too to trigger restart/bootstrap similarly to `remove_peer`?

Reply:
I maybe have to do something about the StatusReplayer here but apart from that it should be ok. If the local peer is the primary image it shouldn't really matter and if the ImageReplayer is currently replaying you don't have to worry about it either because it would not change the actual peer you are replaying to. And if your have not detected yet a primary image, the ImageReplayer will check the new peer on the next start.

The only thing that could maybe be problematic is to add a peer which have a primary image while you already have a remote peer with a primary image to replay to. For the moment I didn't handle this in the BootstrapRequest as it takes the first primary peer that it can find. If the BootstrapRequest logic remains the same I don't think we have to worry about stopping the replay here.